### ISSUE TITLE
Zuul (Dalston.SR5) is forcefully adding Accept-Encoding to gzip

### ISSUE THREAD

**Issue by stiyyagura** [spring-cloud/spring-cloud-netflix#2998]
Bug report

Zuul (Dalston.SR5) is forcefully adding Accept-Encoding to gzip though the request has Accept-Encoding='identity'. Please refer issue #1818 

ProxyRequestHelper.java
```
	public MultiValueMap<String, String> buildZuulRequestHeaders(
			HttpServletRequest request) {
		RequestContext context = RequestContext.getCurrentContext();
		MultiValueMap<String, String> headers = new HttpHeaders();
		Enumeration<String> headerNames = request.getHeaderNames();
		if (headerNames != null) {
			while (headerNames.hasMoreElements()) {
				String name = headerNames.nextElement();
				if (isIncludedHeader(name)) {
					Enumeration<String> values = request.getHeaders(name);
					while (values.hasMoreElements()) {
						String value = values.nextElement();
						headers.add(name, value);
					}
				}
			}
		}
		Map<String, String> zuulRequestHeaders = context.getZuulRequestHeaders();
		for (String header : zuulRequestHeaders.keySet()) {
			headers.set(header, zuulRequestHeaders.get(header));
		}
		headers.set(HttpHeaders.ACCEPT_ENCODING, "gzip");
		return headers;
	}
```

**Comment by ryanjbaxter** [comment]
Fixed via 381a097

**Comment by stiyyagura** [comment]
@ryanjbaxter I still feel we shouldn't default to  Accept-Encoding to gzip this header must only come from the service consumer/caller. Let's take an example where the consumer only want gzipped repose if he ask for it using this header if he is not then we should not send the compressed response. Please let me know your thoughts.
